nupic/algorithms/anomaly.py

```diff
@@ -154,9 +161,33 @@ def compute(self, activeColumns, predictedColumns,
       else:
         score = 0.0
 
+    self._iteration += 1
+    if not self.isReady():
+      score = 0.5
     return score
 
 
+  def reset(self):
+    """
+    reset internal state on sequence end ('r' signal in OPF, 
+    or CLAModel.sequenceReset())
+    """
+    if self._movingAverage is not None:
+      self._movingAverage.reset()
+    if self._likelihood is not None:
+      self._likelihood.reset()
+
+
+  def isReady(self):
+    """
+    @return True when anomaly predictions should be accurate (after burn-in)
+    """
+    bMA = (self._movingAverage is None) or self._movingAverage.isReady()
+    bLike = (self._likelihood is None) or self._likelihood.isReady()
+    bCla = self._iteration > self._burnIn
+    return bCla and bMA and bLike
```

Comment:
What about `len(self._historicalScores) < self._probationaryPeriod` as seen before in https://github.com/numenta/nupic/pull/2267/files#diff-afb5fcf88cb59a3944b9820b5ef3ca5dL133 ?